tests/run/i5340.scala

```diff
@@ -0,0 +1,23 @@
+object Test {
+  def main(args: Array[String]): Unit = {
+    lazy val a: Int = { println("Int"); 1 }
```

Comment:
This test doesn't really check anything. A more useful test could be to check that the bytecode size of the generated methods are under X bytes for some constant X.

Reply:
They exercices all possible type of LazyHolder. I'll keep them and add bytecode tests as well